proto/storage/usage.proto

```diff
@@ -8,7 +8,8 @@ import "google/protobuf/timestamp.proto";
 option go_package = "storage";
 
 message Usage {
-    google.protobuf.Timestamp timestamp = 1 [(gogoproto.moretags) = 'sql:"pk" search:"Timestamp"'];
-    int64 num_nodes                     = 2;
-    int64 num_cpu_units                 = 3;
+    string id                           = 1 [(gogoproto.moretags) = 'sql:"pk,type(uuid)"'];
```

Comment:
I tried to walk the suite of PRs.  This information is collected by cluster id.  It is put in the cache by cluster id.  Then that data is aggregated and stored in the database as a total.  IMO, I think it would make sense to store it in the database by cluster id.  That provides a few benefits, IMO:
1. it gives us an obvious primary key
2. it provides more flexibility as the next logical ask form a customer would be to find out which cluster is using what resources
3. it probably simplifies implementation.  Could we get rid of the cache altogether and just store the data in the database by cluster id as it comes in?

The downside is we would have to aggregate the data that is stored in the DB vs what is in the cache, but that could be as simple as writing some aggregation queries or worst case walk what is in the DB which is exactly what we do with the cache now.

It is possible I'm missing some background and context here, but from a DB perspective it seems more logical to use the database for these functions than to handle it all in memory and then push the aggregations to the DB.

Reply:
I see your point. Clusters send the metrics every 5 minutes. We don't want to store the data per cluster every 5 minutes, so we aggregate data in the cache.


If customers want to track resource usage, they're welcome to look at the telemetry.